app/migrations.go

```diff
@@ -191,34 +191,36 @@ func (a *App) DoGuestRolesCreationMigration() {
 	}
 	for _, scheme := range schemes {
 		if scheme.DefaultTeamGuestRole == "" || scheme.DefaultChannelGuestRole == "" {
-			// Team Guest Role
-			teamGuestRole := &model.Role{
-				Name:          model.NewId(),
-				DisplayName:   fmt.Sprintf("Team Guest Role for Scheme %s", scheme.Name),
-				Permissions:   roles[model.TEAM_GUEST_ROLE_ID].Permissions,
-				SchemeManaged: true,
-			}
-
-			if savedRole, err := a.Srv().Store.Role().Save(teamGuestRole); err != nil {
-				mlog.Critical("Failed to create new guest role for custom scheme.", mlog.Err(err))
-				allSucceeded = false
-			} else {
-				scheme.DefaultTeamGuestRole = savedRole.Name
-			}
-
-			// Channel Guest Role
-			channelGuestRole := &model.Role{
-				Name:          model.NewId(),
-				DisplayName:   fmt.Sprintf("Channel Guest Role for Scheme %s", scheme.Name),
-				Permissions:   roles[model.CHANNEL_GUEST_ROLE_ID].Permissions,
-				SchemeManaged: true,
-			}
-
-			if savedRole, err := a.Srv().Store.Role().Save(channelGuestRole); err != nil {
-				mlog.Critical("Failed to create new guest role for custom scheme.", mlog.Err(err))
-				allSucceeded = false
-			} else {
-				scheme.DefaultChannelGuestRole = savedRole.Name
+			if scheme.Scope == model.SCHEME_SCOPE_TEAM {
+				// Team Guest Role
+				teamGuestRole := &model.Role{
+					Name:          model.NewId(),
+					DisplayName:   fmt.Sprintf("Team Guest Role for Scheme %s", scheme.Name),
+					Permissions:   roles[model.TEAM_GUEST_ROLE_ID].Permissions,
+					SchemeManaged: true,
+				}
+
+				if savedRole, err := a.Srv().Store.Role().Save(teamGuestRole); err != nil {
+					mlog.Critical("Failed to create new guest role for custom scheme.", mlog.Err(err))
+					allSucceeded = false
+				} else {
+					scheme.DefaultTeamGuestRole = savedRole.Name
+				}
+			} else if scheme.Scope == model.SCHEME_SCOPE_CHANNEL {
```

Comment:
I think this `else if` is wrong. If I remember correctly (and If I'm interpreting correctly the code in `store/sqlstore/scheme_store.go:115`), the `DefaultChannelGuestRole` is going to be part any `SCHEME_SCOPE_TEAM` and `SCHEME_SCOPE_CHANNEL` so... the `DefaultTeamGuestRole` is only for the `SCHEME_SCOPE_TEAM` but the `DefaultChannelGuestRole` is for both.

Reply:
That makes sense. That means we only need the check on the team scope.

Reply:
Yes, `SCHEME_SCOPE_TEAM` should get the Channel roles as well. So the `else if` isn't needed.